### PR TITLE
Multi-Z improvements part 2

### LINKED ISSUE #41966: Falling onto space turf deals damage
If you fall down a Z level onto space (/turf/open/space), you will take damage as if you fell down onto a solid floor. Probably also applies to similar turfs.

Mckenzie Fields falls through the open space!x2
Mckenzie Fields crashes into space!
Mckenzie Fields crashes into space with a sickening noise!

### LINKED ISSUE #47068: Falling onto the SM crystal from above does not dust you.
## Reproduction: 
1. Go to the open space one Z level above the SME
2. Make a tile on the tile besides the one directly above the SME
3. Step onto the open space above the SME
4. You are now ontop of the SM crystal without being dusted.

image

### PR DESCRIPTION
**Pull request by TheChosenEvilOne**
## About The Pull Request
Fixes multiple issues with Multi-Z stations and improves the multi-z experience.

fixes #47068
fixes #41966
and fixes some issues mentioned in #47075

## Why It's Good For The Game
Multi-Z is good.

## Changelog
:cl:
fix: You can now pull anything up stairs.
fix: Anchored objects no longer fall down openspace.
fix: Catwalks block falling when on openspace.
fix: Falling on to the SM crystal now dusts you.
fix: Falling on groundless turfs no longer damages you.
tweak: You can use the RCD on openspace now.
tweak: Gravity generator now affects all station z-levels.
/:cl:

### CODE REVIEW

**Comment by HomocidlePancakes** [comment]
Will it trigger if I drop SM crystal on SM?

**Comment by TheChosenEvilOne** [comment]
> 
> 
> Will it trigger if I drop SM crystal on SM?

yes

**Comment by Tlaltecuhtli** [comment]
can you add bouncing carpet that when you fall you get bounced back on the top z level?

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -684,6 +684,21 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
+		AM.visible_message("<span class='danger'>\The [AM] smacks into \the [src] and rapidly flashes to ash.</span>", null,\
+		"<span class='hear'>You hear a loud crack as you are washed with a wave of heat.</span>")
+	else
+		return FALSE
+
+	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 50, TRUE)
+
+	Consume(AM)
+
 /obj/machinery/power/supermatter_crystal/proc/Consume(atom/movable/AM)
```
Comment (actioninja) [inline code/modules/power/supermatter/supermatter.dm]: Isn't this the same as otherwise touching it?

**Review by actioninja — CHANGES_REQUESTED** [review]

File: code/modules/power/supermatter/supermatter.dm (hunk 1 above)
Comment (TheChosenEvilOne) [reply code/modules/power/supermatter/supermatter.dm]: yes, I guess I could just make it call the proc for touching it.

**Comment by Ty-the-Smonk** [comment]
Hulks should be able to punch up and jump up a z level for memes

**Comment by FloranOtten** [comment]
> Hulks should be able to punch up and jump up a z level for memes

They should be able to, but because they jump onto openspace, they instantly fall back down.

**Comment by HomocidlePancakes** [comment]
Can I drop pianos to gib people?

**Pull request merged**